src/Symfony/Component/Yaml/Tests/ParserTest.php

```diff
@@ -1351,6 +1351,30 @@ public function parserThrowsExceptionWithCorrectLineNumberProvider()
             ),
         );
     }
+
+    public function testParsePhpConstant()
+    {
+        $yaml = 'foo: !php/const:PHP_INT_MAX';
+        $php = array('foo' => PHP_INT_MAX);
+
+        $this->assertSame($php, $this->parser->parse($yaml, Yaml::PARSE_CONSTANT));
+    }
+
+    public function testParsePhpConstantThrowsExceptionWhenUndefined()
+    {
+        $expectedMessage = 'The constant "WRONG_CONSTANT" is not defined';
+        $this->setExpectedException('\Symfony\Component\Yaml\Exception\ParseException', $expectedMessage);
+
+        $this->parser->parse('foo: !php/const:WRONG_CONSTANT', Yaml::PARSE_CONSTANT);
+    }
+
+    public function testParsePhpConstantThrowsExceptionOnInvalidType()
+    {
+        $expectedMessage = '#The string "!php/const:PHP_INT_MAX" could not be parsed as constant.*#';
+        $this->setExpectedExceptionRegExp('\Symfony\Component\Yaml\Exception\ParseException', $expectedMessage);
+
+        $this->parser->parse('foo: !php/const:PHP_INT_MAX', Yaml::PARSE_EXCEPTION_ON_INVALID_TYPE);
+    }
```

Comment:
Not sure if it's really necessary to duplicate all the tests from the `Inline` class here.